archive/walk.go

```diff
@@ -38,7 +38,14 @@ func Walk(ark *Archive, visit Visitor) error {
 type SpanVisitor func(si SpanInfo, zardir string) error
 
 func SpanWalk(ark *Archive, v SpanVisitor) error {
-	for _, s := range ark.Spans {
+	if _, err := ark.UpdateCheck(); err != nil {
+		return err
+	}
+
+	ark.mu.RLock()
+	defer ark.mu.RUnlock()
```

Comment:
There's a race here where the spans could be updated in between the `UpdateCheck` and `RLock`. Probably not the end of the world if the SpanWalk is a little stale/out of date, just wanted to make sure this was the intended level of consistency.

Reply:
`... eventually consistent` - yep, exactly